src/components/Topic.jsx

```diff
@@ -22,16 +23,26 @@ function Topic() {
         const response = await api.get(`/articles`, {
           params: {
             topic: topics,
+            sort_by: sortBy,
+            order: order,
+            limit: limit,
+            p: page,
           },
         });
         setArticles(response.data.articles);
+        // if changing the page amount would push user out of range of the pages, put them on the last page
+        const pageCalc = Math.ceil(response.data.total_count / limit);
+        if (pageCalc < page) {
+          setPage(pageCalc);
+        }
+        setPageAmount(pageCalc);
       } catch (err) {
         console.log(err);
       }
     };
 
     fetchArticles();
-  }, [checkedTopics, topic_slug]);
+  }, [checkedTopics, limit, order, page, sortBy, topic_slug]);
```

Comment:
React will throw a warning here because setPage and setPageAmount isn't in the dependancy array.